src/backend/gporca/libgpos/src/common/CStackDescriptor.cpp

```diff
@@ -87,6 +83,57 @@ CStackDescriptor::BackTrace(ULONG top_frames_to_skip)
 		next_frame = (void **) *next_frame;
 	}
 }
+#else
+{
+	// get base pointer of current frame
+
+	ULONG gpos_stack_trace_depth_actual;
+	void *frames[GPOS_STACK_TRACE_DEPTH];
+	gpos_stack_trace_depth_actual = backtrace(frames, GPOS_STACK_TRACE_DEPTH);
+
+	// reset stack depth
+	Reset();
+
+	// get stack start address
+	ULONG_PTR stack_start = 0;
+	IWorker *worker = IWorker::Self();
+	if (NULL == worker)
+	{
+		// no worker in stack, return immediately
+		return;
+	}
+
+	// get address from worker
+	stack_start = worker->GetStackStart();
+
+	// consider the first GPOS_STACK_TRACE_DEPTH frames below worker object
+	for (ULONG frame_counter = 0; frame_counter < gpos_stack_trace_depth_actual;
+		 frame_counter++)
+	{
+		// check if the frame pointer is after stack start and before previous frame
+		// note that stack is *assumed* to be growing *down*
+		if ((ULONG_PTR) frames[frame_counter] > stack_start ||
```

Comment:
@leskin-in backtrace call returns `return addresses` not frame pointers. Why do we compare stack addresses with code addresses?

Reply:
This is leftover code from assembly-based implementation. It ensures the unwiding "progresses". `backtrace()`, however, _guarantees_ correctness of its output, thus we do not need this code.

Reply:
Fixed in https://github.com/arenadata/gpdb/pull/235/commits/7514e7f32f4f9f9b83a38e9babbf024ecdcb664a.